api/test/lib/libDependecies.test.js

```diff
@@ -0,0 +1,32 @@
+beforeEach(() => {
+  jest.resetModules();
+  jest.mock('../../lib/SqlServerConnection');
+  jest.mock('../../lib/PostgresDb', () => jest.fn());
+  jest.mock('../../lib/JigsawUtils.js', () => ({
+    doJigsawGetRequest: jest.fn(),
+    doJigsawPostRequest: jest.fn(),
+    doGetRequest: jest.fn()
+  }));
+
+  jest.mock('../../lib/gateways/SingleView/CustomerLinks', () => () => ({
+    execute: jest.fn(async () => [])
+  }));
+});
+
+describe('libDependencies', () => {
+  it('is able to initialise the use cases', async () => {
+    const {
+      addVulnerability,
+      customerSearch,
+      fetchDocuments,
+      fetchNotes,
+      fetchRecords
+    } = require('../../lib/libDependencies');
+
+    expect(addVulnerability).toBeDefined();
+    expect(customerSearch).toBeDefined();
+    expect(fetchDocuments).toBeDefined();
+    expect(fetchNotes).toBeDefined();
+    expect(fetchRecords).toBeDefined();
```

Comment:
I like this

Reply:
I'm not sure exactly what this achieves - happy to discuss

Reply:
This is an initial test for libDependencies which ensures that all the requires in libDependencies succeed since we added a bunch of new ones. We discussed this with Ben on Friday so happy to implement any alternative suggestions for a basic initial test for libDependencies.

Reply:
I would argue that libDependencies doesn't need any tests - there's no logic there, it's just a big config file basically. What it looks like you are testing is that the files get loaded and some functions get exported which kinda seems overkill to me

Reply:
Hmm. It's pretty easy to require a file that doesn't exist or move a file without coming here to update its require path, esp. during a refactor like we were doing here, without any test failures to notify about the problem currerently.

There's also the issue where the use cases here could get initialised with the wrong gateways (again most likely during a refactor, esp. for new starters). I was wondering whether there's a good way to test that not only are the use cases exported successfully but that they were passed the right gateways on init.

But if it's preferred to have some manual test steps required for this app then I guess best to document those quite precisely somewhere? Otherwise have to remember to tell new starters each time (or whoever on project handover).

Reply:
If a require path is wrong the application will go bang when the require happens, so that's a pretty good failure notification I'd say. If you think these tests serve a purpose then all good, these were just my opinions

Reply:
I guess for me the issue is that the "bang" would only happen once you've pushed to prod as it would be entirely possible to put a typo in here but all tests still pass. If we had integration tests that booted up the actual app then it would be safe and I would argue that this is no longer needed.

Reply:
Agree. I'll add a note to the integration tests ticket ([this one?](https://trello.com/c/LrEcCAaS/52-integration-testing)) that this libDependencies test can be removed as redundant once they're done.